src/Kiota.Builder/Writers/Php/CodeClassDeclarationWriter.cs

```diff
@@ -15,8 +15,8 @@ public override void WriteCodeElement(ClassDeclaration codeElement, LanguageWrit
             if(writer == null) throw new ArgumentNullException(nameof(writer));
             conventions.WritePhpDocumentStart(writer);
             conventions.WriteNamespaceAndImports(codeElement, writer);
-            var derivation = (codeElement?.Inherits == null ? string.Empty : $" extends {codeElement.Inherits.Name.ToFirstCharacterUpperCase()}") +
-                                (!codeElement.Implements.Any() ? string.Empty : $" implements {codeElement.Implements.Select(x => x.Name).Aggregate((x,y) => x + ", " + y)}");
+            var derivation = (codeElement?.Inherits == null ? string.Empty : $" extends {codeElement.Inherits?.Name?.ToFirstCharacterUpperCase()}") +
+                                (!(codeElement.Implements.Any() && (codeElement?.Inherits == null)) ? string.Empty : $" implements {codeElement?.Implements?.Select(x => x?.Name).Aggregate((x,y) => x + ", " + y)}");
```

Comment:
```suggestion
                                (!(codeElement.Implements.Any() && (codeElement.Inherits == null)) ? string.Empty : $" implements {codeElement.Implements.Select(x => x.Name).Aggregate((x,y) => x + ", " + y)}");
```

I'm curious to why inheritance changes anything to implementation statements? (on top of my suggestion I believe the condition is wrong)

Reply:
The goal was to remove an implementation for a child class if the parent already implements it(For cleaner code). But the above approach is wrong in some cases.

Reply:
nothing guarantees that the base class will implement the same interfaces the derived ones need to implement.
E.g. Entity could implement parsable, and directoryObject that derives from it parsable AND additionalDataHolder.